src/build/tenscript/build_phase.rs

```diff
@@ -171,35 +172,37 @@ impl BuildPhase {
         }
     }
 
-    pub fn init(&mut self, fabric: &mut Fabric) {
+    pub fn init(&mut self, fabric: &mut Fabric, brick_library: &impl BrickLibrary) -> Result<(), TenscriptError> {
         let (buds, marks) =
-            Self::execute_node(fabric, Scratch, &self.root, vec![]);
+            Self::execute_node(fabric, Scratch, &self.root, vec![], brick_library)?;
         self.buds = buds;
         self.marks = marks;
+        Ok(())
     }
 
     pub fn is_growing(&self) -> bool {
         !self.buds.is_empty()
     }
 
-    pub fn growth_step(&mut self, fabric: &mut Fabric) {
+    pub fn growth_step(&mut self, fabric: &mut Fabric, brick_library: &impl BrickLibrary) -> Result<(), TenscriptError> {
```

Comment:
I think you can do without the &. Both Library and &Library implement BrickLibrary

Reply:
Seems it gets moved then where it shouldn't.

Reply:
Right okay